### ISSUE TITLE
Limit number of chips / items visible at top of selector

### ISSUE THREAD

**Issue by cciccole** [ng-select/ng-select#1298]
**Is your feature request related to a problem? Please describe.**
This problem is preventing me from using this component.
The problem is that when selecting many items, or when selecting items with long display names, the top of the selector gets huge with a bunch of chips that represent the selected items. This isn't appropriate for in-line display. It also seems implicated in performance issues.

**Describe the solution you'd like**
There should be a configuration that limits the number of chips that get shown at the top. The better implementation may be to just limit to one line with no wrapping, since the number of chips to show will depend on their lengths. The chips shouldn't be inserted into the DOM at all.

When the number of selections is bigger than the allowed space to display the chips, there should be an ellipsis or a small box that indicates the number of non-shown selections.

As a side note, the number of selections should always be indicated somewhere visible. Maybe that should be a separate feature request.

**Describe alternatives you've considered**
The only alternative is to show all the chips, which is the current behavior that is preventing me from using this component.

**Comment by varnastadeus** [comment]
It is already possible, check demo page

**Comment by cciccole** [comment]
Well I'm asking for it to be automatic and based on the space available. Right now I think it requires a bit too much manual configuration for being a basic use case. I think the same thing about select all/none and select all/none for search-filtered items. These make sense to be in the API for the component itself

**Comment by varnastadeus** [comment]
there is already a similar issue regarding automatic implementation

**Comment by sp-daniel-sanchez** [comment]
Hi @varnastadeus @cciccole,

I'm having the same issue right now.
I have a default of 100 elements and it's weird to show all of them at once in the select.

Is there any way to limit the amount of items displayed and show a counter next to them or something like this?

Thanks in advance!

**Comment by cciccole** [comment]
@sp-daniel-sanchez I took the approach of only showing a summary label. For example:

![image](https://user-images.githubusercontent.com/3837064/127246447-586d8466-4e24-464b-ac8e-f31ca12e2a44.png)

Template:
```
<ng-select ...>
  <ng-template *ngIf="multiple" ng-multi-label-tmp let-items="items">
    <div class="ng-value">
      <span class="ng-value-label">{{items.length}} of {{mselect.items.length}} selected</span>
    </div>
  </ng-template>
</ng-select>
```

The reference to `mselect` is this:
```
@ViewChild(NgSelectComponent, { static: true }) mselect: NgSelectComponent;
```

The reference to `multiple` is probably obvious but that is the same argument passed to the `multiple` attribute of the `ng-select`. The selection count label shouldn't be shown if not in multiple mode.

A more complex solution could be developed for showing a certain amount of chips before no longer showing them, but I decided it wasn't worth it for my use case. The most that would ever be shown is 1-3, which offers no value above simply showing the overall count.
